### DISCUSSION TITLE
Confusion about memory usage of cache

### DISCUSSION THREAD

**Discussion by yqz0203** [vercel/swr#1943] [General] [answered]
SWR is a great tool and I want to use it as a low-level data fetching api in my application. As far as I know, SWR will cache data per different key.

So I wonder if a lot of `useSWR` is used, eg. There are many product detail pages in an app, will it cause excessive memory usage?

**Comment by koba04** [comment, marked as answer]
I think it doesn't have to matter in most cases, but It depends on how much you save data into your cache-store. SWR provides a way to create a custom cache. So you can implement an LRU cache if you want.

https://swr.vercel.app/docs/advanced/cache

The following is a related PR for this.
https://github.com/vercel/swr/pull/92

**Reply by yqz0203** [reply]
Great! Thanks.
